metrics.py

```diff
@@ -122,13 +123,9 @@ def evidence_distinct_fields_count(
     melted = melted.withColumn('variable', f.lit(var_name))
     return melted
 
-
-def read_parquet_if_provided(spark, parquet_path):
-    if parquet_path:
-        return spark.read.parquet(parquet_path)
-    else:
-        return None
-
+def read_file_if_provided(spark, filename):
+    if filename:
+        return spark.read.json(filename) if 'json' in filename else spark.read.parquet(filename)
```

Comment:
Maybe something like this? Although TBH I would prefer separate functions for this as this makes things a bit obscure (judgement call).

```suggestion
        return spark.read.json(filename) if filename.endswith('.json') else spark.read.parquet(filename)
```

Reply:
This wouldn't work as most of the JSON files come gzipped. What do you mean by obscure?

Reply:
Then maybe something like `filename.endswith('.json') or filename.endswith('.json.gz')`? I'm just trying to make this check as strict as possible, so that (for example) `json` _somewhere_ in the preceding path doesn't trigger this.

What I mean by obscurity is that, at the point of reading the file, it's not evident in the code which format we're reading from; instead we're using a magical function, which _does_ admittedly handle the things for us, but at a cost of slightly reduced transparency. This is perhaps somewhat similar to what you rightly pointed out in the PhenoDigm parser, where centralised column renames were hiding what was actually happening at the point of loading the files.

**However,** on a second thought I actually think we don't need to change anything here. In an unlikely event that something goes wrong, Spark will let us know by refusing to read the wrong format, so we'll be fine